Platforms/QemuQ35Pkg/Library/ConfigKnobShimLib/ConfigKnobShimLibCommon.c

```diff
@@ -0,0 +1,160 @@
+/** @file
+  Common functionality for the library interface for the autogen XML config header to call into to fetch a config value.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+#include <Library/BaseMemoryLib.h>
+
+#define CONFIG_INCLUDE_CACHE
+#include <Generated/ConfigClientGenerated.h>
+#include <Generated/ConfigServiceGenerated.h>
+
+#define CONFIG_KNOB_NAME_MAX_LENGTH  64

```

Comment:
Knob lengths can be determined from the xml parsing of the CfgData. Would it be useful to move this define into an auto generated file that it output from the parsing?

Reply:
Brent added this :). Looking at it, I think we can get rid of it altogether and simply use the string size * 2 of the ascii string being passed in.